### ISSUE TITLE
Autograd fails with dim set in reduction ops

### ISSUE THREAD

**Issue by skaae** [pytorch/pytorch#1813]
```# runs
x = Variable(torch.ones(5), requires_grad=True)
y  = x.pow(2).mean()
y.backward()

# fails
x = Variable(torch.ones(5), requires_grad=True)
y  = x.pow(2).mean(0)
y.backward()
```
I’m using using version: 0.1.12+ea58190

EDIT: maybe its related to https://github.com/pytorch/pytorch/issues/1787

**Comment by gchanan** [comment]
It's not really related to https://github.com/pytorch/pytorch/issues/1787, because there isn't broadcasting going on here.  What looks it looks like is happening here is keepdim is False (the default), so the forward squeezes and backward unsqueezes.  But in the 1-dimensional case, the squeeze doesn't actually do anything, so the backward is incorrect.  I'll fix it.

**Comment by soumith** [comment]
this is fixed in master
